ReactWindows/ReactNative.Shared/UIManager/BorderedCanvasManager.cs

```diff
@@ -32,37 +38,57 @@ protected override void AddView(TCanvas parent, DependencyObject child, int inde
         /// <param name="parent">The parent view.</param>
         /// <param name="index">The index.</param>
         /// <returns>The child view.</returns>
-        protected override FrameworkElement GetChildAt(TCanvas parent, int index)
+        public override DependencyObject GetChildAt(TCanvas parent, int index)
         {
-            return (FrameworkElement)parent.Children[index];
+            if (HasBorder(parent))
+            {
+                index++;
+            }
+
+            return parent.Children[index];
         }
 
         /// <summary>
         /// Gets the number of children in the view parent.
         /// </summary>
         /// <param name="parent">The view parent.</param>
         /// <returns>The number of children.</returns>
-        protected override int GetChildCount(TCanvas parent)
+        public override int GetChildCount(TCanvas parent)
         {
-            return parent.Children.Count;
+            var count = parent.Children.Count;
+
+            if (HasBorder(parent))
+            {
+                count--;
+            }
+
+            return count;
         }
 
         /// <summary>
         /// Removes all children from the view parent.
         /// </summary>
         /// <param name="parent">The view parent.</param>
-        protected override void RemoveAllChildren(TCanvas parent)
+        public override void RemoveAllChildren(TCanvas parent)
         {
-            parent.Children.Clear();
+            for (var i = GetChildCount(parent) - 1; i >= 0; i--)
+            {
+                RemoveChildAt(parent, i);
+            }
```

Comment:
Can you inject a HasBorder check and just clear children in case there is no border? That way there's no net performance impact on views without a border.